src/secret_sharing/xor.rs

```diff
@@ -20,3 +26,155 @@ impl XorReplicated {
         self.right
     }
 }
+
+/// XOR secret share.
+///
+/// Bits are stored in the Little-Endian format. Accessing the first element
+/// like `s[0]` will return the LSB.
+///
+/// ## Example
+/// ```
+/// use raw_ipa::secret_sharing::XorSecretShare;
+///
+/// let s = XorSecretShare::<64>::from(2_u128);
+/// let b0 = s[0];
+/// let b1 = s[1];
+///
+/// assert_eq!(b0, 0);
+/// assert_eq!(b1, 1);
+/// ```
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct XorSecretShare<const N: usize>(BitVec<u8, Lsb0>);
+
+// BitVec uses `usize` as its default store type in memory. Using `u8`
+// here allows us to store bits larger than 32 but less than 64 most
+// effectively (i.e., 40-bit = 5 * `u8` blocks).
+
+impl<const N: usize> XorSecretShare<N> {
+    const BITS: usize = N;
+    const SIZE_IN_BYTES: usize = Self::BITS / 8;
+
+    #[allow(dead_code)]
+    #[must_use]
+    pub fn new(v: BitVec<u8, Lsb0>) -> Self {
```

Comment:
doesn't it need two values because it is secret share? also asking callers to prepare a bit array/vec is rough, you may want to make it private and provide convenient `From` implementation

Reply:
Per offline discussion, this is intended to not have left/right for this struct. This represents a single secret-shared value. I already have `From` implementation, so I'll delete `new`. The field is already private.